python/change-notes/2021-02-10-yaml-more-loading-functions.md

```diff
@@ -0,0 +1,2 @@
+lgtm,codescanning
+* Improved modeling for the `PyYAML` PyPI package (imported as `yaml`), now supporting `safe_load`, `unsafe_load`, and `full_load` (as well as the `..._load_all` functions). In the current version of PyYAML (5.4.1), only `safe_load` is known to not be safe from code execution exploits. Calls to the other functions are now modeled as sinks of the _Deserializing untrusted input_ (`py/unsafe-deserialization`) query.
```

Comment:
Would this change work, to simplify "known to not be safe from":

```suggestion
* Improved modeling for the `PyYAML` PyPI package (imported as `yaml`), now supporting `safe_load`, `unsafe_load`, and `full_load` (as well as the `..._load_all` functions). In the current version of PyYAML (5.4.1), only `safe_load` is known to be vulnerable to code execution exploits. Calls to the other functions are now modeled as sinks of the _Deserializing untrusted input_ (`py/unsafe-deserialization`) query.
```

Reply:
Thanks, I must have rewritten that once too many, and ended up absolutely botching it :laughing: The only one that is safe is `safe_load`, which is hopefully what the change note also says now :laughing:

Reply:
Thanks for the update. Clearly I should also have been concentrating harder when I reviewed too.

Content all LGTM now 👍🏻